ci/builders/linux_framework_smoke.json

```diff
@@ -0,0 +1,70 @@
+{
+    "builds": [
+        {
+            "archives": [
+                {
+                    "name": "host_debug_unopt",
+                    "base_path": "out/host_debug_unopt/zip_archives/",
+                    "type": "gcs",
+                    "include_paths": [
+                        "out/host_debug_unopt/zip_archives/dart-sdk-linux-x64.zip"
+                    ],
+                    "realm": "production"
+                }
+            ],
+            "drone_dimensions": [
+                "device_type=none",
+                "os=Linux"
+            ],
+            "gn": [
+                "--unoptimized",
+                "--prebuilt-dart-sdk"
+            ],
+            "name": "host_debug_unopt",
+            "ninja": {
+                "config": "host_debug_unopt",
+                "targets": [
+                    "flutter/build/archives:dart_sdk_archive"
+                ]
+            }
+        }
+    ],
+    "tests": [
+        {
+            "name": "test: lint host_debug widgets",
```

Comment:
It would be really helpful to have comments or links to the test commands that this will actually run. One advantage of the engine v1 recipe was that it was totally unambiguous what commands would be run. For example here: https://flutter.googlesource.com/recipes/+/refs/heads/main/recipes/engine/framework_smoke.py#85

Reply:
For example, it would be great if the config here was self-documenting, and it was obvious how to add more tests. Right now, these test configs are more-or-less full of magic strings, and it's not clear where to mine more magic strings from.

Reply:
Json does not support comments. To be able to add comments the configs will need to get migrated to yaml or proto. To mitigate this I added docs to the README file in build configuration directory.

The magic strings are coming from the framework test runner. Unfortunately it has no owner, no structure and zero documentation.

Reply:
> Json does not support comments. To be able to add comments the configs will need to get migrated to yaml or proto. To mitigate this I added docs to the README file in build configuration directory.

This is usually accomplished by adding non-interpreted fields like `"_comment"`.

> The magic strings are coming from the framework test runner. Unfortunately it has no owner, no structure and zero documentation.

@keyonghan How did you find the strings that are in this PR?

Reply:
To be honest, this is not a trivial thing even though I have been relatively familiar with ci.yaml, recipes, and some parts of test runners..

This is how I did:

1. Identify the tests from v1 framework smoke tests: `flutter analyze` + `flutter test`
2. For `flutter analyze`
    1. First search across `flutter` repository test runners for `analyze` related tests, no good match 
    2. Then search existing `flutter` post-submit tests/builders/targets which are related to `analyze`, find `Linux analyzer_benchmark` and `Linux analyze`
    3. Looking into these two `analyze` tests, I find `Linux analyze` is the closest, where `flutter analyze --flutter-repo` is covered.
    4. Based on the recipes which `Linux analyze` is using, I find `validation` related properties/strings
3. For `flutter test`
    1. Based on the path this test runs against, `pacakges/flutter`, search related codes in existing test runners (dev/bots/test.dart)
    2. Find related shards: `framework_tests`, and subshards: `widget`, `libraries`, `mise` and `slow`.
    3. Then look through the detailed list of the tests `flutter test` runs, and match to what those subshards cover
    4. Finalize `widget` and `libraries`, which cover all the tests in v1.
 
As @godofredoc mentions, missing ownership of these test runners makes it difficult to achieve this.

Reply:
I don't think what I'm asking has anything to do with ownership of the tests.

All I want to know is where do these strings come from.

For example, it looks like below, the `"shard"`, and `"subshard"` fields come from the framework `.ci.yaml` file here:

https://github.com/flutter/flutter/blob/master/.ci.yaml#L498.

Can you please add a note somewhere in this file that the `"shard"` and `"subshard"` fields come from the framework `.ci.yaml` file?

Reply:
Added a comment for both shard and validation.

Reply:
Thanks!